iothub_client/samples/pnp/pnp_temperature_controller/pnp_temperature_controller.c

```diff
@@ -470,24 +521,34 @@ static IOTHUB_DEVICE_CLIENT_LL_HANDLE CreateDeviceClientAndAllocateComponents(vo
 
     if (result == false)
     {
-        PnP_ThermostatComponent_Destroy(g_thermostatHandle2);
         PnP_ThermostatComponent_Destroy(g_thermostatHandle1);
-        if (deviceClient != NULL)
-        {
-            IoTHubDeviceClient_LL_Destroy(deviceClient);
-            IoTHub_Deinit();
-            deviceClient = NULL;
-        }
+        PnP_ThermostatComponent_Destroy(g_thermostatHandle2);
     }
 
-    return deviceClient;
+    return result;
 }
 
 int main(void)
 {
     IOTHUB_DEVICE_CLIENT_LL_HANDLE deviceClient = NULL;
 
-    if ((deviceClient = CreateDeviceClientAndAllocateComponents()) == NULL)
+    g_pnpDeviceConfiguration.modelId = g_temperatureControllerModelId;
+    g_pnpDeviceConfiguration.enableTracing = g_hubClientTraceEnabled;
```

Comment:
should this come from Environment setting?

Reply:
We could, but I *think* it's OK as-is.  The other stuff like conn strings/DPS scopeIds we want as environment vars because we don't want people hard-coding secrets in .c files and this makes experimentation a bit easier.  Generally when you (the developer) compile a sample EXE though you have a good sense of whether you want it with full debug logs blasting (which is the default since we're not that verbose) or not.

Also complicating this is we'd need to coordinate potentially with all the non-embedded SDKs, which follow this pattern too.

So my inclination is to won't-fix this, though if we make future SDK work that's totaly fresh we should reconsider this and anything else that could be from env.